src/nvim/api/private/helpers.c

```diff
@@ -604,6 +605,9 @@ void modify_keymap(Buffer buffer, bool is_unmap, String mode, String lhs, String
     return;
   }
 
+  if (opts != NULL && opts->func.type == kObjectTypeLuaRef) {
```

Comment:
>And this is a helper function for api not an api function, I'm not sure what you meant by it will always be defined per the API contract.

Yep, you're right! This was the part I missed in my review :)